notebooks/kaggle/store-sales-time-series-forecasting/experiments/linear_regression_v2.ipynb

```diff
@@ -0,0 +1,3133 @@
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "os.chdir(os.environ['PROJECT_ROOT'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e2443dc6",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2022-08-22T08:57:18.680600Z",
+     "start_time": "2022-08-22T08:57:17.625106Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib.dates as mdates\n",
+    "import seaborn as sns\n",
+    "import sklearn\n",
+    "import pdpipe as pdp\n",
+    "from pandas.core.common import SettingWithCopyWarning\n",
+    "from sklearn.base import BaseEstimator, RegressorMixin, MetaEstimatorMixin, TransformerMixin, clone\n",
+    "from datetime import timedelta\n",
+    "from statistics import median, mean, stdev\n",
+    "from pdpipe import df\n",
+    "from pathlib import Path\n",
+    "from sklearn.linear_model import LinearRegression, Ridge, Lasso\n",
+    "from sklearn.preprocessing import OneHotEncoder, MinMaxScaler\n",
+    "from sklearn.pipeline import make_pipeline\n",
+    "from sklearn.model_selection import cross_validate, cross_val_score, TimeSeriesSplit, GridSearchCV\n",
+    "from sklearn.compose import ColumnTransformer\n",
+    "from sklearn.multioutput import MultiOutputRegressor, RegressorChain\n",
+    "from sklearn.metrics import mean_squared_log_error, mean_squared_error, mean_absolute_error, r2_score, mean_absolute_percentage_error\n",
+    "from mentorship.ml.models.reg import PositiveRegressor\n",
+    "from mentorship.ml.models.common import SplitPipeline\n",
+    "from mentorship.ml.models.kaggle.storesales.linear import PipelineLinearV1\n",
+    "from mentorship.ml.models.kaggle.storesales.ridge import PipelineRidgeV1\n",
+    "from mentorship.ml.models.kaggle.storesales.lasso import PipelineLassoV1\n",
+    "from mentorship.ml.models.kaggle.storesales.elasticnet import PipelineElasticNetV1\n",
+    "from mentorship.features.kaggle.storesales.etl import ETLTransformer\n",
+    "from mentorship.ml.cv.split import DateTimeSeriesSplit\n",
+    "from mentorship.ml.cv.util import print_cv_test_scores\n",
+    "\n",
+    "\n",
+    "%matplotlib inline\n",
+    "warnings.simplefilter(action='ignore', category=FutureWarning)\n",
+    "warnings.simplefilter(action=\"ignore\", category=SettingWithCopyWarning)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "18239ede",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2022-08-22T08:57:23.324283Z",
+     "start_time": "2022-08-22T08:57:23.310607Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "CV_METRICS = [\n",
+    "    'neg_mean_squared_log_error',\n",
+    "    'neg_root_mean_squared_error',\n",
+    "    'neg_mean_absolute_error',\n",
+    "    # 'neg_mean_absolute_percentage_error',\n",
```

Comment:
I read some articles about this problem and mostly it is said that the best option is to change the metric and don't use MAPE, but there were some other solutions:
1) change MAPE for SMAPE, which is pretty similar to MAPE (https://en.wikipedia.org/wiki/Symmetric_mean_absolute_percentage_error). This metric can deal with zero actual target values, but it won't work if both actual and predicted values are equal to 0. There are some modifications of this metric (also listed in wiki), but such modifications won't be so similar to MAPE;
2) change the data: for example, adding 1 to every sample target. But I think, that is not a good option, because it will cause changes in the given dataset and not trustful CV scores;
3) remove samples in the data where target is equal to 0. This is also not a good option, because in my situation there are a lot of zeros in the target column, and removing such a big number of samples also won't give us trustful CV results.